src/SourceGenerators/Uno.UI.SourceGenerators/Helpers/HashBuilder.cs

```diff
@@ -28,25 +45,29 @@ public static string BuildIDFromSymbol(ITypeSymbol typeSymbol)
 		/// <param name="input">The string to hash</param>
 		/// <param name="algoritm">The algorithm to use</param>
 		/// <returns>A hex-formatted string of the hash</returns>
-		public static string Build(string input, HashAlgorithm algoritm = null)
+		public static string Build(string input)
 		{
-			using (algoritm ??= SHA256.Create())
+			using (var algoritm = SHA256.Create())
 			{
 				// Convert the input string to a byte array and compute the hash.
 				var data = algoritm.ComputeHash(Encoding.UTF8.GetBytes(input));
 
-				// Create a new Stringbuilder to collect the bytes
-				// and create a string.
-				var builder = new StringBuilder();
-
-				// Use the first 16 bytes of the hash
-				for (int i = 0; i < Math.Min(data.Length, 16); i++)
+				var min = Math.Min(data.Length, 16);
+				var result = new string((char)0, min * 2);
```

Comment:
Missed this before, this is undefined behavior and you should definitely never do this. The best you can do on .NET Standard 2.0 is renting a `char[]`  from the pool, writing data to it, then slicing as a span and calling `ToString()`, then returning the array to the pool. But you should never, for any reason, mutate data from an existing `string` instance.

Reply:
@Sergio0694 Why not use Span directly instead of renting an array?

```
        [Benchmark]
        public string RentArr()
        {
            using (var algorithm = SHA256.Create())
            {
                var data = algorithm.ComputeHash(Encoding.UTF8.GetBytes(Input));

                var min = Math.Min(data.Length, 16);
                var array = ArrayPool<char>.Shared.Rent(min * 2);
                var span = array.AsSpan();
                unsafe
                {
                    for (int i = 0; i < min; i++)
                    {
                        var s = formattedBytes[data[i]];
                        span[i * 2] = s[0];
                        span[i * 2 + 1] = s[1];
                    }
                }

                ArrayPool<char>.Shared.Return(array);
                return span.ToString();
            }
        }

        [Benchmark]
        public string UseSpan()
        {
            using (var algorithm = SHA256.Create())
            {
                var data = algorithm.ComputeHash(Encoding.UTF8.GetBytes(Input));

                var min = Math.Min(data.Length, 16);
                
                Span<char> span = stackalloc char[min * 2];
                unsafe
                {
                    for (int i = 0; i < min; i++)
                    {
                        var s = formattedBytes[data[i]];
                        span[i * 2] = s[0];
                        span[i * 2 + 1] = s[1];
                    }
                }

                
                return span.ToString();
            }
        }
```

Performance-wise, the difference is negligible (using span directly is ~1.473 us, and renting array is ~1.563 us when running the benchmark on my machine). Is renting preferred for some reason?

Reply:
If the length has an upper bound, sure, using `stackalloc` is fine. Just hardcode it to 32 though.
Also you can drop the `unsafe` block entirely then 🙂